lib/addons/src/transitional-decorator.js

```diff
@@ -0,0 +1,40 @@
+import deprecate from 'util-deprecate';
+
+// Create a decorator that can be used both in the (deprecated) old "hoc" style:
+//   .add('story', decorator(options)(() => <Story />));
+//
+// And in the new, "parameterized" style:
+//   .addDecorator(decorator)
+//   .add('story', () => <Story />, { name: { parameters } });
+
+export const makeTransitionalDecorator = ({ name, parameterName, wrapper }) => {
```

Comment:
Should we rename it to something like `makePolymorphicDecorator`?

Reply:
Sure! Or just `makeDecorator()`?

Reply:
I like the makeDecorator name too